employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/pages/rotation/ShiftTemplateEditForm.java

```diff
@@ -0,0 +1,242 @@
+                Duration newDurationOfTemplate;
+                if (to.getDayOffset() >= from.getDayOffset()) {
+                    newDurationOfTemplate = Duration.ofDays(to.getDayOffset()).plusSeconds(to.getTime().toSecondOfDay()).minus(newDurationFromReference);
+                } else {
+                    newDurationOfTemplate = Duration.ofDays(to.getRotationLength()).minus(newDurationFromReference).plusDays(to.getDayOffset()).plusSeconds(to.getTime().toSecondOfDay());
+                }
+                template.setDurationBetweenRotationStartAndTemplateStart(newDurationFromReference);
+                template.setShiftTemplateDuration(newDurationOfTemplate);
+            } catch (ValidationException invalidField) {
+                template.setRotationEmployeeId(oldEmployee);
+                template.setDurationBetweenRotationStartAndTemplateStart(oldDurationFromReference);
+                template.setShiftTemplateDuration(oldDurationOfTemplate);
+                return;
+            }
+
+            RotationRestServiceBuilder.updateShiftTemplate(template.getTenantId(), template, FailureShownRestCallback.onSuccess((final ShiftTemplateView updatedShiftTemplate) -> {
+                shiftTemplateGridObject.getShiftTemplateModel().withShiftTemplateView(updatedShiftTemplate);
+                shiftTemplateGridObject.getElement().classList.remove(SELECTED_CLASS);
+                eventManager.fireEvent(ROTATION_INVALIDATE);
+                hide();
+            }).onFailure(i -> {
+                template.setRotationEmployeeId(oldEmployee);
+                template.setDurationBetweenRotationStartAndTemplateStart(oldDurationFromReference);
+                template.setShiftTemplateDuration(oldDurationOfTemplate);
+            }).onError(i -> {
+                template.setRotationEmployeeId(oldEmployee);
+                template.setDurationBetweenRotationStartAndTemplateStart(oldDurationFromReference);
+                template.setShiftTemplateDuration(oldDurationOfTemplate);
+            }));
+        } else {
+            final ShiftTemplateView template = new ShiftTemplateView();
+
+            template.setRotationEmployeeId(parseId(employeeSelect.getSelectedValue()));
+            Duration newDurationFromReference = Duration.ofDays(from.getDayOffset()).plusSeconds(from.getTime().toSecondOfDay());
+            Duration newDurationOfTemplate;
+            if (to.getDayOffset() >= from.getDayOffset()) {
+                newDurationOfTemplate = Duration.ofDays(to.getDayOffset()).plusSeconds(to.getTime().toSecondOfDay()).minus(newDurationFromReference);
+            } else {
+                newDurationOfTemplate = Duration.ofDays(to.getRotationLength()).minus(newDurationFromReference).plusDays(to.getDayOffset()).plusSeconds(to.getTime().toSecondOfDay());
+            }
+            template.setDurationBetweenRotationStartAndTemplateStart(newDurationFromReference);
+            template.setShiftTemplateDuration(newDurationOfTemplate);
```

Comment:
Lines 203-212 seem to be duplicate of lines 169-178. Could you extract the duplicate code into a method and reuse it? I believe it will improve readability as well.

Reply:
Thanks; refactored